### PR TITLE
crossref-harvester: add parser for the crossref events format

### PR DESCRIPTION
**Pull request by Glignos**

### CODE REVIEW

**Comment by Glignos** [comment]
By processing the events from crossref we can give them a working "scholix" like format.
There is an issue that we should consider though before moving further on this: The events endpoint of crossref is not providing us with the type of the publications, the obvious solution to this would be assigning it the default value `unknown`, but this is bound to change the processing behavior.

**Comment by slint** [comment]
Leaving the type as `unknown` is fine, since the PIDs will get harvested for metadata afterwards.

File: asclepias_broker/harvester/crossref.py [hunk 1]
```diff
@@ -92,6 +92,35 @@ def _clean_scholix(self, data):
             data[k]['Identifier']['IDURL'] = data[k]['Identifier'].pop('IDUrl')
         return data
 
+    def _parse_doi_from_url(self, url):
```
Comment (slint) [inline asclepias_broker/harvester/crossref.py]: `idutils` can do that

File: asclepias_broker/harvester/crossref.py [hunk 2]
```diff
@@ -92,6 +92,35 @@ def _clean_scholix(self, data):
...
+        """."""
+        splitted_url = url.split('/')
+        doi = splitted_url[-2] + '/' + splitted_url[-1]
+        return doi
+
+    def _transform_events(self, data):
```
Comment (slint) [inline asclepias_broker/harvester/crossref.py]: This has to be called depending on if the events are Scholix or not

Comment (Glignos) [reply asclepias_broker/harvester/crossref.py]: Yup I haven't connected it yet since I was expecting feedback.

**Pull request merged**